CMakeLists.txt

```diff
@@ -82,15 +82,15 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
-  set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -ggdb -DDEBUG")
-  set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
-  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-g -ggdb -O2")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -DSPFFT_SINGLE_PRECISION=ON")
```

Comment:
This should not be added to the compile flags, since this is a CMake flag, that should be used when configuring SpFFT to support single precision.

Reply:
Can we check inside the host code if `SPFFT_SINGLE_PRECISION` is enabled? Then we can check if we can work with FP32 FFTs or not.

Reply:
Yes, the CMake config file of SpFFT defines it. So after `find_package(SpFFT 1.0.2 CONFIG REQUIRED)`, `SPFFT_SINGLE_PRECISION` is available to check. It is also defined as a macro, which is included in the `spfft.hpp` header. I'd still recommend adding an option, to catch any issue if someone explicitly requires it (it also helps with package managers like Spack). The option values could be `auto`, `on` or `off`, and default to `auto`, where we enable single precision if possible.

Reply:
> I'd still recommend adding an option, 
Option on the Sirius side? For example, introduce `SIRIUS_SINGLE_PRECISION` in CMakeLists.txt?

Reply:
Yes. Although, I guess to keep it similar to the current options, maybe `USE_SINGLE_PRECISION`?